### PR TITLE
Reorder boxmodes for flight mode priority

### PR DESCRIPTION
**Pull request by breadoven**
Reorders boxmodes to prioritise flight modes within the first 32 entries. This ensures flight modes and more significant modes are included in blackbox flight logs.

Numbering has been removed from the `boxId_e` enum since it doesn't appear to serve any useful purpose.

Doesn't appear to cause any issues from testing ...

Change affects Blackbox Explorer which will need updating to ensure correct boxmode mapping is used depending on INAV version (Ref Blackbox change https://github.com/iNavFlight/blackbox-log-viewer/pull/101).

### CODE REVIEW

**Comment by stronnag** [comment]
Why not add an additional field to blackbox?

And will you also provide a PR for [INAV blackbox-tools](https://github.com/iNavFlight/blackbox-tools) as required?

**Comment by breadoven** [comment]
> Why not add an additional field to blackbox?

Do you mean a separate field for actual "flight" modes and another field for all other modes ?

> And will you also provide a PR for [INAV blackbox-tools](https://github.com/iNavFlight/blackbox-tools) as required?

I have something that works. Not sure if it's the best way of doing it ... but it works.

**Comment by stronnag** [comment]
> > Why not add an additional field to blackbox?
> 
> Do you mean a separate field for actual "flight" modes and another field for all other modes ?

I was thinking of maybe an `flightModeFlags2` which would be the remaining modes right shifted 32 to fit into another `uint32_t`; but maybe that's overkill (and a pain on the consumer side).

> > And will you also provide a PR for [INAV blackbox-tools](https://github.com/iNavFlight/blackbox-tools) as required?
> 
> I have something that works. Not sure if it's the best way of doing it ... but it works.

There is an example in `src/parser.c` from then the VBAT stuff changed in the middle of the development cycle (so version and data dependent). Or just ping me when this is merged and I'll do it.

**Comment by mmosca** [comment]
Will this impact 7->8 upgrades? If so, we should mention it in the release notes.

File: src/main/fc/rc_modes.h [hunk 1]
```diff
@@ -26,61 +26,61 @@
...
-    BOXUSER4         = 49,
-    BOXCHANGEMISSION = 50,
-    BOXBEEPERMUTE    = 51,
-    BOXMULTIFUNCTION = 52,
-    BOXMIXERPROFILE      = 53,
-    BOXMIXERTRANSITION   = 54,
-    BOXANGLEHOLD     = 55,
+    BOXARM,
+    BOXANGLE,
+    BOXHORIZON,
```
Comment (mmosca) [inline src/main/fc/rc_modes.h:31]: Having the explicit numbers for the enumeration is nice for referencing the code to know what each of the modes are.

Comment (DzikuVx) [reply src/main/fc/rc_modes.h:31]: I agree with that, more useful than counting

**Comment by breadoven** [comment]
> Will this impact 7->8 upgrades? If so, we should mention it in the release notes.

Shouldn't have any impact on users other than certain modes selections such as Cruise being recorded in log files.

File: src/main/fc/rc_modes.h (hunk 1 above)
Comment (breadoven) [reply src/main/fc/rc_modes.h:31]: I guess it's useful for debugging, which will only return the number, in which case commented numbering make more sense, let the enum apply the sequential numbering itself surely.

**Comment by breadoven** [comment]
Replaced by #10093.

**Pull request closed without merging**